presto-product-tests/src/main/java/io/prestosql/tests/hive/TestHiveViews.java

```diff
@@ -295,6 +297,40 @@ public void testSelectFromHiveViewWithoutDefaultCatalogAndSchema()
                 .containsExactly(row(1L));
     }
 
+    @Test
+    public void testTimestampHiveView() {
+        onHive().executeQuery("DROP TABLE IF EXISTS timestamp_hive_table");
+        onHive().executeQuery("CREATE TABLE timestamp_hive_table AS SELECT cast('1990-01-02 12:13:14.123456789' AS timestamp) ts");
+        onHive().executeQuery("DROP VIEW IF EXISTS timestamp_hive_view");
+        onHive().executeQuery("CREATE VIEW timestamp_hive_view AS SELECT * FROM timestamp_hive_table");
+
+        // timestamp_precision not set UNSET
+        assertThat(query("SELECT CAST(ts as VARCHAR) FROM timestamp_hive_view")).containsExactly(row("1990-01-02 12:13:14.123"));
```

Comment:
as -> AS
VARCHAR -> varchar

(we tend to write SQL keywords in uppercase and identifiers in lowercase)